rslib/linkchecker/tests/links.rs

```diff
@@ -1,94 +1,137 @@
+use linkcheck::validation::Context;
+use linkcheck::validation::Reason;
+use linkcheck::BasicContext;
+use regex::Regex;
+use reqwest::Url;
+use strum::IntoEnumIterator;
 
-    use super::*;
+/// Aggregates [`Outcome`]s by collecting the error messages of the invalid
+/// ones.
+#[derive(Default)]
+struct Outcomes(Vec<String>);
 
-    /// Aggregates [`Outcome`]s by collecting the error messages of the invalid
-    /// ones.
-    #[derive(Default)]
-    struct Outcomes(Vec<String>);
+enum Outcome {
+    Valid,
+    Invalid(String),
+}
 
-    enum Outcome {
-        Valid,
-        Invalid(String),
-    }
+#[derive(Clone)]
+enum CheckableUrl {
+    HelpPage(HelpPage),
+    String(&'static str),
+}
 
-    #[tokio::test]
-    async fn check_links() {
-        if env::var("ONLINE_TESTS").is_err() {
-            println!("test disabled; ONLINE_TESTS not set");
-            return;
+impl CheckableUrl {
+    fn url(&self) -> Cow<str> {
+        match *self {
+            Self::HelpPage(page) => page.to_link().into(),
+            Self::String(s) => s.into(),
         }
-        let ctx = BasicContext::default();
-        let result = futures::stream::iter(HelpPage::iter())
-            .map(|page| check_page(page, &ctx))
-            .buffer_unordered(ctx.concurrency())
-            .collect::<Outcomes>()
-            .await;
-        if !result.0.is_empty() {
-            panic!("{}", result.message());
+    }
+
+    fn anchor(&self) -> Cow<str> {
+        match *self {
+            Self::HelpPage(page) => page.to_link_suffix().into(),
+            Self::String(s) => s.split('#').last().unwrap_or_default().into(),
         }
     }
+}
+
+impl From<HelpPage> for CheckableUrl {
+    fn from(value: HelpPage) -> Self {
+        Self::HelpPage(value)
+    }
+}
 
-    async fn check_page(page: HelpPage, ctx: &BasicContext) -> Outcome {
-        let link = page.to_link();
-        match Url::parse(&link) {
-            Ok(url) => {
-                if url.as_str() == link {
-                    match check_web(&url, ctx).await {
-                        Ok(()) => Outcome::Valid,
-                        Err(Reason::Dom) => Outcome::Invalid(format!(
-                            "'#{}' not found on '{}{}'",
-                            url.fragment().unwrap(),
-                            url.domain().unwrap(),
-                            url.path(),
-                        )),
-                        Err(Reason::Web(err)) => Outcome::Invalid(err.to_string()),
-                        _ => unreachable!(),
-                    }
-                } else {
-                    Outcome::Invalid(format!(
-                        "'{}' is not a valid URL part",
-                        page.to_link_suffix(),
-                    ))
+impl From<&'static str> for CheckableUrl {
+    fn from(value: &'static str) -> Self {
+        Self::String(value)
+    }
+}
+
+fn ts_help_pages() -> impl Iterator<Item = &'static str> {
+    lazy_static! {
```

Comment:
No need to change this, but just FYI: I've recently started using once_cell's Lazy instead of lazy_static, as it seems to be the preferred solution these days, and may one day end up in the Rust stdlib.